provisioning/hare_setup.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Setup utility for Hare to configure Hare related settings, e.g. logrotate,
+# report unsupported features, etc.
+
+import argparse
+import asyncio
+import json
+import logging
+import shutil
+import subprocess
+import os
+import yaml
+
+from cortx.utils.product_features import unsupported_features
+
+def execute_command_and_get_output(cmd):
```

Comment:
`def execute(cmd: str) -> Tuple[str, int]` is enough.

Btw do you really need to return exit code? The thing is that normally I don't care what was the exit code. Let it just raise an exception if exit code was not 0. So the caller doesn't need to re-check exit code. If exit code is not good, there will be an exception (so the error is not swallowed), but if the things are fine (exit code = 0), then I don't need to analyze it specially. I'll just concentrate on positive scenario in the code.

Reply:
DOone

Reply:
@SwapnilGaonkar7, you its done but somehow I don't see the updated code. Also please check the codacy warning about `returncode`.
As the matter of fact please see if you can address the rest of the codacy warnings/errors too.
Thank you.

Reply:
@mssawant @knekrasov I have fixed above issue and also fixed few codacy warnings/errors.
I have also done some modification in 'checkRpm' function.
Please have a look